File: array/unique_elements.cpp
// Problem: Given a sorted list of integers in ascending order, return the number of unique elements k and the array with the first k elements being the unique element
// Approach: Start with index n = 1 instead of 0, compare it with preceding element, in case of unique elements, move it forward in the array

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // handle empty or array with size 1
        if (nums.size() <= 1)
        {
            return nums.size();
        }
        // else traverse vector
        int k = 1;
        for(int i = 1;i< nums.size();i++)
        {
            //check if not equal with previous element
            if(nums[i] != nums[i-1])
            {
                nums[k] = nums[i];
                //increment k since unique element
                ++k;
            }
        }
        return k;
    }
};

File: array/uniqueElements.java
// Problem name: Remove duplicates from sorted array II
// Problem desc: Given a sorted array in ascending order where at most an element can appear twice, 
// return the array with the first k unique elements 
// Approach: Same with sorted array I but with k = 2 since an element is allowed to appear twice

class Solution {
    public int removeDuplicates(int[] nums) {
        // handle array with less than 2 elements
        if(nums.length <= 2){
            return nums.length;
        }
        // else traverse array
        int k = 2;
        for(int i=2;i<nums.length;i++)
        {
            if(nums[i] != nums[k-2])
            {
                nums[k] = nums[i];
                ++k;
            }
        }
        return k;
    }
}
